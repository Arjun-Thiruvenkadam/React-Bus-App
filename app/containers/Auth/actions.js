/*
 *
 * Auth actions
 *
 */

import * as actionTypes from './constants';

export function inputChange(event, controlName) {
  return {
    type: actionTypes.INPUT_CHANGE,
    event,
    controlName,
  };
}

export function submit(event) {
  return {
    type: actionTypes.SUBMIT,
    event,
  };
}

export function startLoading() {
  return {
    type: actionTypes.START_LOADING,
  };
}

export function stopLoading() {
  return {
    type: actionTypes.STOP_LOADING,
  };
}

export function authSuccess(name, token) {
  return {
    type: actionTypes.AUTH_SUCCESS,
    name,
    token,
  };
}

export function switchAuthMode(){
  return{
    type: actionTypes.SWITCH_AUTHMODE,
  }
}

export function logout(){
  return{
    type: actionTypes.LOGOUT,
  }
}

export function invalid(){
  return{
    type: actionTypes.INVALID,
  }
}
