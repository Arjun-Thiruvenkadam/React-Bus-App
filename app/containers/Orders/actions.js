/*
 *
 * Orders actions
 *
 */

import * as actionTypes from './constants';

export function bookTicket() {
  return {
    type: actionTypes.BOOK_TICKET,
  };
}

export function showSpinner() {
  return {
    type: actionTypes.SHOW_SPINNER,
  };
}

export function hideSpinner() {
  return {
    type: actionTypes.HIDE_SPINNER,
  };
}

export function saveTickets(tickets) {
  return {
    type: actionTypes.SAVE_TICKETS,
    tickets,
  };
}

export function cancelOrder() {
  return {
    type: actionTypes.CANCEL_ORDER,
  };
}

export function selectTicket(ticketId) {
  return {
    type: actionTypes.SELECT_TICKET,
    ticketId,
  };
}

export function fetchData() {
  return {
    type: actionTypes.FETCH_DATA,
  };
}

export function completeBooking() {
  return {
    type: actionTypes.COMPLETE_BOOKING,
  };
}

export function clearSelection() {
  return {
    type: actionTypes.CLEAR_SELECTION,
  };
}

export function update() {
  return {
    type: actionTypes.UPDATE,
  };
}
