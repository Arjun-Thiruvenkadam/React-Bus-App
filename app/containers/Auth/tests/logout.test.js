import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { browserHistory } from 'react-router-dom';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import Logout, { mapDispatchToProps } from '../logout';
import * as actions from '../actions';
import configureStore from '../../../configureStore';

describe('<Logout />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <MemoryRouter>
          <IntlProvider locale="en">
            <Logout />
          </IntlProvider>
        </MemoryRouter>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    describe('logoutHandler', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.logoutHandler).toBeDefined();
      });

      it('should dispatch logout when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.logoutHandler();
        expect(dispatch).toHaveBeenCalledWith(actions.logout());
      });
    });
  });
});
