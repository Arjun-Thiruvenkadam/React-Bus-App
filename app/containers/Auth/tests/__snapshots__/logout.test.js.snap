// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logout /> should render and match the snapshot 1`] = `null`;
